### PR TITLE
Relocated mod methods to API

### PR DESCRIPTION
**Pull request by IGN-Styly**
Relocates base methods to the API for addon use.

### Contributor Liscence Agreement
[Full CLA](https://github.com/iron431/Irons-Spells-n-Spellbooks/blob/1.19.2/CLA.md)
*Contributors: Please "X" in the following box to acknowledge our CLA*
- [X] I have read and agree to the CLA for Iron's Spells and Spellbooks which allows me to retain my ownership in the code submitted while granting the repository owner legal rights to use my contribution.

### CODE REVIEW

**Comment by IGN-Styly** [comment]
Also removes need for the UpdateClient utility class, by sending the mana update event via Messages.

File: src/main/java/io/redspace/ironsspellbooks/api/magic/MagicData.java
Comment (iron431) [inline src/main/java/io/redspace/ironsspellbooks/api/magic/MagicData.java:1]: none of these changes on magic data should be made

**Review by iron431 — CHANGES_REQUESTED** [review]
also this is in the wrong branch, 1.19.2 is the development branch; 1.20.1 should not be worked on directly

File: src/main/java/io/redspace/ironsspellbooks/api/magic/MagicData.java
Comment (iron431) [reply src/main/java/io/redspace/ironsspellbooks/api/magic/MagicData.java:1]: to explain better, sending the packet here bypasses optimization we've made to reduce network traffic (and still left in the manual packet syncing), and there seem to be erroneous imports added to the file

**Comment by IGN-Styly** [comment]
sent to wrong repo my bad

**Pull request closed without merging**